### PR TITLE
Micro se linux

### PR DESCRIPTION
**Pull request by lvicoun**
### PR creator: Description

Describe the overall goals of this pull request.

### PR creator: Are there any relevant issues/feature requests?

* bsc#...
* jsc#SLE-...

### PR creator: Which product versions do the changes apply to?

When opening a PR, check all versions of the documentation that your PR applies to.

- SLE 15/openSUSE Leap 15.x
  - [ ] SLE 15 SP4/openSUSE Leap 15.4 *(current `main`, no backport necessary)*
  - [ ] SLE 15 SP3/openSUSE Leap 15.3
  - [ ] SLE 15 SP2/openSUSE Leap 15.2
  - [ ] SLE 15 SP1
  - [ ] SLE 15 GA
- SLE 12
  - [ ] SLE 12 SP5
  - [ ] SLE 12 SP4
  - [ ] SLE 12 SP3

### PR reviewer only: Have all backports been applied?

The doc team member merging your PR will take care of backporting to older documents.
When opening a PR, do *not* set the following check box.

- [ ] all necessary backports are done

### CODE REVIEW

File: xml/art_admin_slemicro_selinux.xml [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+  <title>Introduction to SELinux</title>
+  <para>
+   SELinux was developed as an additional Linux security solution that uses the
+   security framework in the Linux kernel. The purpose was to allow for a more
+   granular security policy that goes beyond the standard Discretionary Access
+   Controls (DAC), the traditional file permissions of owner/group/world, and
+   read/write/execute.
+  </para>
+  <para>
+   SELinux uses labels attached to objects (for example, files, binaries, and
```
Comment (jsegitz) [inline xml/art_admin_slemicro_selinux.xml:22]: - binaries. They're just files

File: xml/art_admin_slemicro_selinux.xml [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+   network sockets) and uses them to determine privilege boundaries, thereby
+   building up a level of confinement that can span more than a process, or
+   even the whole system.
```
Comment (jsegitz) [inline xml/art_admin_slemicro_selinux.xml:25]: and uses them for access decisions.  Remove the rest. SELinux doesn't really determine privilege boundaries, it enforces them

File: xml/art_admin_slemicro_selinux.xml [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+  </para>
+  <para>
+   SELinux was developed in accordance with some certification standards like
+   Common Criteria and FIPS 140.
```
Comment (jsegitz) [inline xml/art_admin_slemicro_selinux.xml:29]: drop that, that's not correct

File: xml/art_admin_slemicro_selinux.xml [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+  </para>
+  <para>
+   SELinux default action is to deny any access that would violate rules
+   defined by the policy. Other feature of SELinux that increases security is
```
Comment (jsegitz) [inline xml/art_admin_slemicro_selinux.xml:33]: Better: "SELinux default action is to deny any access. Only those actions, that were specifically allowed can happen"

File: xml/art_admin_slemicro_selinux.xml [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+   that processes run within their own domain and cannot access files of other
+   processes that run in different domains. A process cannot access another
```
Comment (jsegitz) [inline xml/art_admin_slemicro_selinux.xml:35]: That's not strictly true, so I would rephrase this to something like "SELinux allows strict confinement of processes up to the point where they can't access the files of other processes on the same machine anymore"

File: xml/art_admin_slemicro_selinux.xml [hunk 6]
```diff
@@ -0,0 +1,353 @@
...
+   process as well, unless configured otherwise.
+  </para>
+  <para>
+   SELinux was designed to enhance existing security solutions not to replace
+   them, for example, discretionary access control is still applied even
+   if the system is using SELinux. If DAC denies access first, SELinux policies
+   are not applied.
```
Comment (jsegitz) [inline xml/art_admin_slemicro_selinux.xml:42]: rephrase this to emphasize that SELinux doesn't get involved. The policy is still applied, it's just not used in this moment

File: xml/art_admin_slemicro_selinux.xml [hunk 7]
```diff
@@ -0,0 +1,353 @@
...
+   labelled.
+  </para>
+  <para>
+   To set the SELinux mode to <literal>enforcing</literal> and configure your
+   system properly for using SELinux, run the following command:
+  </para>
+<screen>&prompt.root;transactional-update setup-selinux</screen>
+  <para>
+   Reboot your system after the command completes. The command installs the
+   SELinux policy if it is not installed, sets the SELinux mode and rebuilts
```
Comment (jsegitz) [inline xml/art_admin_slemicro_selinux.xml:62]: sets it to what?

File: xml/art_admin_slemicro_selinux.xml [hunk 8]
```diff
@@ -0,0 +1,353 @@
...
+  </para>
+  <para>
+    &slema; si shipped with the <literal>targeted</literal> SELinux policy.
+  </para>
+ </section>
+ <section xml:id="sec-security-context">
+  <title>SELinux security context</title>
+  <para>
+   The security context is a set of information assigned to a file or a
+   process. It consists of identity, role, type and level. This informaton is
```
Comment (jsegitz) [inline xml/art_admin_slemicro_selinux.xml:173]: s/identity/SELinux user/

File: xml/art_admin_slemicro_selinux.xml [hunk 9]
```diff
@@ -0,0 +1,353 @@
...
+   used to make access control decisions.
+  </para>
+  <variablelist>
+   <title></title>
+   <varlistentry>
+    <term>identity</term>
```
Comment (jsegitz) [inline xml/art_admin_slemicro_selinux.xml:179]: same

File: xml/art_admin_slemicro_selinux.xml [hunk 10]
```diff
@@ -0,0 +1,353 @@
...
+     </para>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term>type</term>
+    <listitem>
+     <para>
+      conveys information how particular files and processes can interact. A
+      process is assigned files with a concrete SELinux type and it cannot
+      acces files outside of this type. By convention, the type name is
```
Comment (jsegitz) [inline xml/art_admin_slemicro_selinux.xml:209]: acces*s*

File: xml/art_admin_slemicro_selinux.xml [hunk 11]
```diff
@@ -0,0 +1,353 @@
...
+      <literal>var_t</literal>.
+     </para>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term>level</term>
+    <listitem>
+     <para>
+      is an optional attribute that specifies the range of levels of clearance
+      in the multilevel security.
```
Comment (jsegitz) [inline xml/art_admin_slemicro_selinux.xml:220]: a security context usually has five fields:
unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023
level is described here, but categories are missing

File: xml/art_admin_slemicro_selinux.xml [hunk 12]
```diff
@@ -0,0 +1,353 @@
...
+Policy MLS status:            enabled
+Policy deny_unknown status:   allowed
+Memory protection checking:   requested (insecure)
+Max kernel policy version:    31            
+          </screen>
+    </listitem>
+   </varlistentry>
+  </variablelist>
+  <note>
+   <title>The <literal>Z</literal> option available to other commands</title>
```
Comment (jsegitz) [inline xml/art_admin_slemicro_selinux.xml:346]: The title is repeated directly after that and reads a bit strange

**Review by jsegitz — CHANGES_REQUESTED** [review]

File: xml/art_admin_slemicro_selinux.xml [hunk 13]
```diff
@@ -0,0 +1,344 @@
...
+    <filename>/etc/selinux/.relabelled</filename> to prevent relabelling on
+    subsequent reboots.
+   </para>
+  </note>
+ </section>
+ <section xml:id="sec-policy-overview">
+  <title>SELinux policy overview</title>
+  <para>
+   The policy is the key component in SELinux. Your SELinux policy defines
+   rules that specify which objects can access which files, directories, ports,
```
Comment (dariavladykina) [inline xml/art_admin_slemicro_selinux.xml:140]:
```suggestion
   rules that specify which objects can access specific files, directories, ports,
```

File: xml/art_admin_slemicro_selinux.xml [hunk 14]
```diff
@@ -0,0 +1,344 @@
...
+  </para>
+  <para>
+   When compiling the policy for your system, you will have a choice to either
+   work with a modular policy, or a monolithic policy, where one huge policy is
+   used to protect everything on your system. It is strongly recommended to use
+   a modular policy and not a monolithic policy. Modular policies are much
+   easier to manage.
+  </para>
+  <para>
+    &slema; si shipped with the <literal>targeted</literal> SELinux policy.
```
Comment (dariavladykina) [inline xml/art_admin_slemicro_selinux.xml:157]:
```suggestion
    &slema; is shipped with the <literal>targeted</literal> SELinux policy.
```

File: xml/art_admin_slemicro_selinux.xml [hunk 15]
```diff
@@ -0,0 +1,344 @@
...
+      example: <literal>system_r</literal>.
+     </para>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term>type</term>
+    <listitem>
+     <para>
+      conveys information how particular files and processes can interact. A
+      process is assigned files with a concrete SELinux type and it cannot
```
Comment (dariavladykina) [inline xml/art_admin_slemicro_selinux.xml:199]:
```suggestion
      process is assigned to files with a concrete SELinux type, and it cannot
```

File: xml/art_admin_slemicro_selinux.xml [hunk 16]
```diff
@@ -0,0 +1,344 @@
...
+      displays the current SELinux mode
+     </para>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term><command>fixfiles</command></term>
+    <listitem>
+     <para>
+      enables you to check for issues with a mismatched security context and
+      then fix it
```
Comment (dariavladykina) [inline xml/art_admin_slemicro_selinux.xml:253]:
```suggestion
      then fix them
```

File: xml/art_admin_slemicro_selinux.xml [hunk 17]
```diff
@@ -0,0 +1,344 @@
...
+  <para>
+   SELinux uses labels attached to objects (for example, files and
+   network sockets) and uses them for access decisions.
+  </para>
+  <para>
+   SELinux default action is to deny any access. SELinux allows only actions that were specifically allowed in the SELinux policy. Other feature of SELinux that increases security is
+   that SELinux allows strict confinement of processes up to the point, where the processes cannot access files of other processes on the same system. 
+  </para>
+  <para>
+   SELinux was designed to enhance existing security solutions not to replace
```
Comment (dariavladykina) [inline xml/art_admin_slemicro_selinux.xml:30]:
```suggestion
   SELinux was designed to enhance existing security solutions, not to replace
```

File: xml/art_admin_slemicro_selinux.xml [hunk 18]
```diff
@@ -0,0 +1,344 @@
...
+   them, for example, discretionary access control is still applied even
```
Comment (dariavladykina) [inline xml/art_admin_slemicro_selinux.xml:31]:
```suggestion
   them. For example, discretionary access control is still applied even
```

File: xml/art_admin_slemicro_selinux.xml [hunk 19]
```diff
@@ -0,0 +1,344 @@
...
+   if the system is using SELinux. If DAC denies access first, SELinux policies
+   are not applied.
+  </para>
+ </section>
+ <section xml:id="sec-getting-selinux">
+  <title>Getting SELinux</title>
+  <para>
+   SELinux is installed by default when installing &slema; by &yast; or is part of the
+   pre-built images. In systems installed manually by &yast;, SELinux is in the
+   <literal>disabled</literal> mode by default and the file system is not
```
Comment (dariavladykina) [inline xml/art_admin_slemicro_selinux.xml:41]:
```suggestion
   <literal>disabled</literal> mode by default, and the file system is not
```

File: xml/art_admin_slemicro_selinux.xml [hunk 20]
```diff
@@ -0,0 +1,344 @@
...
+   labelled. If you deployed your system using the pre-built images, the
```
Comment (dariavladykina) [inline xml/art_admin_slemicro_selinux.xml:42]:
```suggestion
   labeled. If you deployed your system using the pre-built images, the
```

File: xml/art_admin_slemicro_selinux.xml [hunk 21]
```diff
@@ -0,0 +1,344 @@
...
+   <literal>permissive</literal>, or <literal>enforcing</literal>.
+  </para>
+  <para>
+   Using the <literal>disabled</literal> means that no rules from the SELinux
+   policy are applied and your system is not protected. Therefore, the
+   <literal>disabled</literal> mode is not recommended.
+  </para>
+  <para>
+   In the <literal>permissive</literal> mode, SELinux is active, the security
+   policy is loaded, the file system is labelled and access denial entries are
```
Comment (dariavladykina) [inline xml/art_admin_slemicro_selinux.xml:70]:
```suggestion
   policy is loaded, the file system is labeled, and access denial entries are
```

File: xml/art_admin_slemicro_selinux.xml [hunk 22]
```diff
@@ -0,0 +1,344 @@
...
+   <filename>/etc/selinux/config</filename> change the value of
+   <literal>SELINUX</literal> to <literal>disabled</literal>, or
+   <literal>permissive</literal>, or <literal>enforced</literal> as follows:
+  </para>
+<screen>SELINUX=disabled</screen>
+  <para>
+   The changes in the file are applied after next reboot.
+  </para>
+  <note>
+   <title>Relabelling your system after switching from the <literal>disabled</literal> mode</title>
```
Comment (dariavladykina) [inline xml/art_admin_slemicro_selinux.xml:116]:
```suggestion
   <title>Relabeling your system after switching from the <literal>disabled</literal> mode</title>
```

File: xml/art_admin_slemicro_selinux.xml [hunk 23]
```diff
@@ -0,0 +1,344 @@
...
+   <para>
+    If you disable SELinux on your system and then enable it later, make sure
+    that you relabel your system. When SELinux is disabled and you perform
+    changes to your file system, the changes are not reflected in the context
+    any more (for example, new files do not have any context), therefore you
+    need to relabel your system by using the <command>restorecon</command>
+    command, using the <literal>autorelabel</literal> boot parameter or by
+    creating a file that will trigger relabelling on the next boot. To create
```
Comment (dariavladykina) [inline xml/art_admin_slemicro_selinux.xml:124]:
```suggestion
    creating a file that will trigger relabeling on the next boot. To create
```

File: xml/art_admin_slemicro_selinux.xml [hunk 24]
```diff
@@ -0,0 +1,344 @@
...
+    the file, run the folowing command:
+   </para>
+<screen>&prompt.root;touch /etc/selinux/.autorelabel</screen>
+   <para>
+    After reboot, the file <filename>/etc/selinux/.autorelabel</filename> is
+    replaced with another flag file:
+    <filename>/etc/selinux/.relabelled</filename> to prevent relabelling on
```
Comment (dariavladykina) [inline xml/art_admin_slemicro_selinux.xml:131]:
```suggestion
    <filename>/etc/selinux/.relabelled</filename> to prevent relabeling on
```

File: xml/art_admin_slemicro_selinux.xml [hunk 25]
```diff
@@ -0,0 +1,344 @@
...
+ </section>
+ <section xml:id="sec-getting-selinux">
+  <title>Getting SELinux</title>
+  <para>
+   SELinux is installed by default when installing &slema; by &yast; or is part of the
+   pre-built images. In systems installed manually by &yast;, SELinux is in the
+   <literal>disabled</literal> mode by default and the file system is not
+   labelled. If you deployed your system using the pre-built images, the
+   SELinux mode is <literal>permissive</literal> and the file system is
+   labelled.
```
Comment (dariavladykina) [inline xml/art_admin_slemicro_selinux.xml:44]:
```suggestion
   labeled.
```

File: xml/art_admin_slemicro_selinux.xml [hunk 26]
```diff
@@ -0,0 +1,344 @@
...
+   denied.
+  </para>
+  <para>
+   In the <literal>enforced</literal> mode, the security policy is applied.
+   Each access that is not explicitly allowed by the policy is denied.
+  </para>
+  <para>
+   You can switch between the <literal>enforcing</literal> and
+   <literal>permissive</literal> modes by using the
+   <command>setenforce</command> command. Or you can switch between all SELinux
```
Comment (dariavladykina) [inline xml/art_admin_slemicro_selinux.xml:81]:
```suggestion
   <command>setenforce</command> command. Alternatively, you can switch between all SELinux
```

File: xml/art_admin_slemicro_selinux.xml [hunk 27]
```diff
@@ -0,0 +1,344 @@
...
+  </para>
+  <para>
+    &slema; si shipped with the <literal>targeted</literal> SELinux policy.
+  </para>
+ </section>
+ <section xml:id="sec-security-context">
+  <title>SELinux security context</title>
+  <para>
+   The security context is a set of information assigned to a file or a
+   process. It consists of SELinux user, role, type, level and category. This informaton is
```
Comment (dariavladykina) [inline xml/art_admin_slemicro_selinux.xml:164]:
```suggestion
   process. It consists of SELinux user, role, type, level and category. This information is
```

File: xml/art_admin_slemicro_selinux.xml [hunk 28]
```diff
@@ -0,0 +1,344 @@
...
+      convention, the role name is suffixed with <literal>_r</literal>, for
+      example: <literal>system_r</literal>.
+     </para>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term>type</term>
+    <listitem>
+     <para>
+      conveys information how particular files and processes can interact. A
```
Comment (dariavladykina) [inline xml/art_admin_slemicro_selinux.xml:198]:
```suggestion
      conveys information on how particular files and processes can interact. A
```

File: xml/art_admin_slemicro_selinux.xml [hunk 29]
```diff
@@ -0,0 +1,344 @@
...
+Memory protection checking:   requested (insecure)
+Max kernel policy version:    31            
+          </screen>
+    </listitem>
+   </varlistentry>
+  </variablelist>
+  <note>
+   <title>The <literal>Z</literal> option available to other commands</title>
+   <para>
+    You can use the <literal>Z</literal> option also with other commands, for
```
Comment (dariavladykina) [inline xml/art_admin_slemicro_selinux.xml:339]:
```suggestion
    You can also use the <literal>Z</literal> option with other commands, for
```

**Review by dariavladykina — COMMENTED** [review]
Hi Jana, 
Please take a look at some changes/comments I made.
Thanks!

File: xml/art_admin_slemicro_selinux.xml [hunk 30]
```diff
@@ -0,0 +1,344 @@
...
+   <literal>permissive</literal> mode or <replaceable>1</replaceable> for the <literal>enforced</literal>
+   mode.
+  </para>
+  <para>
+   To verify the mode, run the following command.
+  </para>
+<screen>&prompt.root;getenforce</screen>
+  <para>
+   The command should return <literal>permissive</literal> or
+   <literal>enforced</literal>, depending on the provided
```
[the remark this answers is not in the document]
Comment (lvicoun) [reply xml/art_admin_slemicro_selinux.xml:102]: The command does not return the mode, it just return one of two possible values. I think that you will see the difference in the output :)

File: xml/art_admin_slemicro_selinux.xml [hunk 31]
```diff
@@ -0,0 +1,344 @@
...
+ </section>
+ <section xml:id="sec-security-context">
+  <title>SELinux security context</title>
+  <para>
+   The security context is a set of information assigned to a file or a
+   process. It consists of SELinux user, role, type, level and category. This informaton is
+   used to make access control decisions.
+  </para>
+  <variablelist>
+   <title></title>
```
[the remark this answers is not in the document]
Comment (lvicoun) [reply xml/art_admin_slemicro_selinux.xml:168]: Yes, added.

File: xml/art_admin_slemicro_selinux.xml [hunk 32]
```diff
@@ -196,7 +196,7 @@
     <listitem>
      <para>
       conveys information how particular files and processes can interact. A
-      process is assigned files with a concrete SELinux type and it cannot
+      process is assigned files with a concrete SELinux type, and it cannot
```
Comment (dariavladykina) [inline xml/art_admin_slemicro_selinux.xml:199]:
```suggestion
      process is assigned to files with a concrete SELinux type, and it cannot
```

**Review by dariavladykina — COMMENTED** [review]
One more small correction.

File: xml/art_admin_slemicro_selinux.xml (hunk 32 above)
Comment (lvicoun) [reply xml/art_admin_slemicro_selinux.xml:199]: It seems my comment got lost, but a process is not assigned to files, but files are assigned to a process. The direction here is the opposite, then suggested by the correction.

File: xml/art_admin_slemicro_selinux.xml (hunk 13 above)
Comment (lvicoun) [reply xml/art_admin_slemicro_selinux.xml:140]: I think that changing wording in this case would change the meaning a little bit. Those files do not necessarily must be specific to the object, they could be assigned to more objects.

File: xml/art_admin_slemicro_selinux.xml [hunk 33]
```diff
@@ -0,0 +1,352 @@
...
+   granular security policy that goes beyond the standard Discretionary Access
+   Controls (DAC), the traditional file permissions of owner/group/world, and
+   read/write/execute.
+  </para>
+  <para>
+   SELinux uses labels attached to objects (for example, files and
+   network sockets) and uses them for access decisions.
+  </para>
+  <para>
+   SELinux default action is to deny any access. SELinux allows only actions that were specifically allowed in the SELinux policy. Other feature of SELinux that increases security is
```
Comment (dariavladykina) [inline xml/art_admin_slemicro_selinux.xml:26]:
```suggestion
   SELinux default action is to deny any access. SELinux allows only actions that were specifically allowed in the SELinux policy. Another feature of SELinux that increases security is
```

File: xml/art_admin_slemicro_selinux.xml [hunk 34]
```diff
@@ -0,0 +1,352 @@
...
+  </para>
+ </section>
+ <section xml:id="sec-getting-selinux">
+  <title>Getting SELinux</title>
+  <para>
+   SELinux is installed by default when installing &slema; by &yast; or is part of the
+   pre-built images. In systems installed manually by &yast;, SELinux is in the
+   <literal>disabled</literal> mode by default, and the file system is not
+   labeled. If you deployed your system using the pre-built images, the
+   SELinux mode is <literal>permissive</literal> and the file system is
```
Comment (dariavladykina) [inline xml/art_admin_slemicro_selinux.xml:43]:
```suggestion
   SELinux mode is <literal>permissive</literal>, and the file system is
```

File: xml/art_admin_slemicro_selinux.xml [hunk 35]
```diff
@@ -0,0 +1,352 @@
...
+     </para>
+<screen>&prompt.root;semanage --help</screen>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term><command>setenforce</command></term>
+    <listitem>
+     <para>
+      enables you to temporarily set a SELinux mode to
+      <literal>permissive</literal> or <literal>enforcing</literal>.
```
Comment (dariavladykina) [inline xml/art_admin_slemicro_selinux.xml:318]:
```suggestion
      <literal>permissive</literal> or <literal>enforcing</literal>
```

File: xml/art_admin_slemicro_selinux.xml [hunk 36]
```diff
@@ -0,0 +1,352 @@
...
+   To verify the mode, run the following command:
+  </para>
+<screen>&prompt.root;getenforce</screen>
+  <para>
+   The command should return <literal>permissive</literal> or
+   <literal>enforced</literal>, depending on the provided
+   <replaceable>MODE_ID</replaceable>.
+  </para>
+  <para>
+   To change the SELiux mode permanently, in the file
```
Comment (dariavladykina) [inline xml/art_admin_slemicro_selinux.xml:106]:
```suggestion
   To change the SELinux mode permanently, in the file
```

**Review by dariavladykina — COMMENTED** [review]
A couple more comments here, sorry :-)
